distributed/utils_test.py

```diff
@@ -788,14 +793,21 @@ async def test_foo():
     )
 
     def _(func):
-        def test_func():
+        @functools.wraps(func)
+        def test_func(*outer_args, **kwargs):
             with clean() as loop:
-                if iscoroutinefunction(func):
-                    cor = func
-                else:
-                    cor = gen.coroutine(func)
```

Comment:
Am I right to assume that since I created the `async def coro` function, we are good to remove this logic from here?

Reply:
Yeah, I think so

Reply:
Awesome thank you

Reply:
Hrm, I'm not as sure that we can remove this logic. In https://github.com/dask/distributed/pull/5551 I'm proposing we slightly modify some existing `gen_test` tests and I noticed that, with the changes in this PR, the updated tests in https://github.com/dask/distributed/pull/5551 fail

Reply:
Thank you for pointing this out, I was having issues getting this part to work and passing the `outer_args` and `kwargs` to the function. I was getting error because we couldn't assign to coroutines. That's the reason why I created that coro function instead.

I suspect that breaks if the `gen_coroutine(coro)` part right? Let me poke at it and see what I can do about it 👍